src/controller/jobmonitor/monitor.go

```diff
@@ -0,0 +1,142 @@
+//  Copyright Project Harbor Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package jobmonitor
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/goharbor/harbor/src/lib/log"
+
+	"github.com/gocraft/work"
+
+	"github.com/goharbor/harbor/src/common/job"
+	"github.com/goharbor/harbor/src/lib/errors"
+	"github.com/goharbor/harbor/src/lib/q"
+	libRedis "github.com/goharbor/harbor/src/lib/redis"
+	jm "github.com/goharbor/harbor/src/pkg/jobmonitor"
+	"github.com/goharbor/harbor/src/pkg/scheduler"
+	"github.com/goharbor/harbor/src/pkg/task"
+)
+
+// All the jobs in the pool, or all pools
+const All = "all"
+
+// Ctl the controller instance of the worker pool controller
+var Ctl = NewWorkerPoolController()
+
+// WorkerPoolController defines the worker pool operations
+type WorkerPoolController interface {
+	// ListPools lists the worker pools
+	ListPools(ctx context.Context) ([]*jm.WorkerPool, error)
+	// ListWorkers lists the workers in the pool
+	ListWorkers(ctx context.Context, poolID string) ([]*jm.Worker, error)
+	// StopRunningJob stop the running job
+	StopRunningJob(ctx context.Context, jobID string) error
+}
+
+type workerPoolController struct {
+	poolManager   jm.PoolManager
```

Comment:
let's call it workerpoolManager? In this PR, we only should have workerpool and worker, and no definition of pool.